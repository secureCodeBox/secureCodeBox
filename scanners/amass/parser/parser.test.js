// SPDX-FileCopyrightText: the secureCodeBox authors
//
// SPDX-License-Identifier: Apache-2.0

const fs = require("fs");
const util = require("util");
const {
  validateParser,
} = require("@securecodebox/parser-sdk-nodejs/parser-utils");

// eslint-disable-next-line security/detect-non-literal-fs-filename
const readFile = util.promisify(fs.readFile);

const { parse } = require("./parser");

test("example parser parses empty json files to zero findings", async () => {
  const fileContent = await readFile(__dirname + "/__testFiles__/empty.jsonl", {
    encoding: "utf8",
  });

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toEqual([]);
});

// test("example parser parses missing json files to zero findings", async () => {
//   expect(await parse(null)).toEqual([]);
// });

// test("example parser parses missing json files to zero findings", async () => {
//   expect(await parse(0)).toEqual([]);
// });

test("example parser parses single line json successfully", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/example.com.jsonl",
    {
      encoding: "utf8",
    }
  );
  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();

  expect(findings).toMatchInlineSnapshot(`
[
  {
    "attributes": {
      "addresses": [
        {
          "asn": 34011,
          "cidr": "10.110.224.0/21",
          "desc": "GD-EMEA-DC-CGN1",
          "ip": "10.110.225.135",
        },
      ],
      "domain": "example.de",
      "name": "www.example.de",
      "source": undefined,
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain www.example.de",
    "location": "www.example.de",
    "name": "www.example.de",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`);
});

test("example parser parses large json result successfully", async () => {
  const fileContent = await readFile(
    __dirname + "/__testFiles__/securecodebox.io.jsonl",
    {
      encoding: "utf8",
    }
  );

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});

// axios parses jsonl with a single line / entry as a json object as they are coincidentally also valid json objects.
// This means that the parser needs to also handle objects passed into into it, not just strings
test("handles jsonl files with a single row correctly", async () => {
  const fileContent = {
    name: "www.securecodebox.io",
    domain: "securecodebox.io",
    addresses: [
      {
        ip: "185.199.109.153",
        cidr: "185.199.108.0/22",
        asn: 54113,
        desc: "FASTLY - Fastly",
      },
      // ...
    ],
    tag: "cert",
    sources: ["Crtsh"],
  };

  const findings = await parse(fileContent);
  await expect(validateParser(findings)).resolves.toBeUndefined();
  expect(findings).toMatchSnapshot();
});
