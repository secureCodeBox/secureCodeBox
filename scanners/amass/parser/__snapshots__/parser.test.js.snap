// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`example parser parses large json result successfully 1`] = `
[
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "grafana.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain grafana.securecodebox.io",
    "location": "grafana.securecodebox.io",
    "name": "grafana.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "kibana.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain kibana.securecodebox.io",
    "location": "kibana.securecodebox.io",
    "name": "kibana.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "build.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain build.securecodebox.io",
    "location": "build.securecodebox.io",
    "name": "build.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "monitoring.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain monitoring.securecodebox.io",
    "location": "monitoring.securecodebox.io",
    "name": "monitoring.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "ui.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain ui.securecodebox.io",
    "location": "ui.securecodebox.io",
    "name": "ui.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "juiceshop.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain juiceshop.securecodebox.io",
    "location": "juiceshop.securecodebox.io",
    "name": "juiceshop.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "wpscan.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain wpscan.securecodebox.io",
    "location": "wpscan.securecodebox.io",
    "name": "wpscan.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "hack.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain hack.securecodebox.io",
    "location": "hack.securecodebox.io",
    "name": "hack.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "spider.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain spider.securecodebox.io",
    "location": "spider.securecodebox.io",
    "name": "spider.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "www.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain www.securecodebox.io",
    "location": "www.securecodebox.io",
    "name": "www.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "rancher.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain rancher.securecodebox.io",
    "location": "rancher.securecodebox.io",
    "name": "rancher.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "backend.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain backend.securecodebox.io",
    "location": "backend.securecodebox.io",
    "name": "backend.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "zap.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain zap.securecodebox.io",
    "location": "zap.securecodebox.io",
    "name": "zap.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "bodgeit.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain bodgeit.securecodebox.io",
    "location": "bodgeit.securecodebox.io",
    "name": "bodgeit.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "api-management.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain api-management.securecodebox.io",
    "location": "api-management.securecodebox.io",
    "name": "api-management.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "dvwa.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain dvwa.securecodebox.io",
    "location": "dvwa.securecodebox.io",
    "name": "dvwa.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "prometheus.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain prometheus.securecodebox.io",
    "location": "prometheus.securecodebox.io",
    "name": "prometheus.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "engine.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain engine.securecodebox.io",
    "location": "engine.securecodebox.io",
    "name": "engine.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "logging.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain logging.securecodebox.io",
    "location": "logging.securecodebox.io",
    "name": "logging.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "petstoreapi.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain petstoreapi.securecodebox.io",
    "location": "petstoreapi.securecodebox.io",
    "name": "petstoreapi.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "jenkins.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain jenkins.securecodebox.io",
    "location": "jenkins.securecodebox.io",
    "name": "jenkins.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "bridge.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain bridge.securecodebox.io",
    "location": "bridge.securecodebox.io",
    "name": "bridge.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "petstore.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain petstore.securecodebox.io",
    "location": "petstore.securecodebox.io",
    "name": "petstore.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "sslyze.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain sslyze.securecodebox.io",
    "location": "sslyze.securecodebox.io",
    "name": "sslyze.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "scanner.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain scanner.securecodebox.io",
    "location": "scanner.securecodebox.io",
    "name": "scanner.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "sieve.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain sieve.securecodebox.io",
    "location": "sieve.securecodebox.io",
    "name": "sieve.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "securecodebox.io",
      "source": "Forward DNS",
      "tag": "dns",
    },
    "category": "Subdomain",
    "description": "Found subdomain securecodebox.io",
    "location": "securecodebox.io",
    "name": "securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "gateway.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain gateway.securecodebox.io",
    "location": "gateway.securecodebox.io",
    "name": "gateway.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "sso.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain sso.securecodebox.io",
    "location": "sso.securecodebox.io",
    "name": "sso.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "test.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain test.securecodebox.io",
    "location": "test.securecodebox.io",
    "name": "test.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "elasticsearch.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain elasticsearch.securecodebox.io",
    "location": "elasticsearch.securecodebox.io",
    "name": "elasticsearch.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "arachni.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain arachni.securecodebox.io",
    "location": "arachni.securecodebox.io",
    "name": "arachni.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "ctf.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain ctf.securecodebox.io",
    "location": "ctf.securecodebox.io",
    "name": "ctf.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "nmap.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain nmap.securecodebox.io",
    "location": "nmap.securecodebox.io",
    "name": "nmap.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "discovery.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain discovery.securecodebox.io",
    "location": "discovery.securecodebox.io",
    "name": "discovery.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "demo.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain demo.securecodebox.io",
    "location": "demo.securecodebox.io",
    "name": "demo.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "dashboard.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain dashboard.securecodebox.io",
    "location": "dashboard.securecodebox.io",
    "name": "dashboard.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "consul.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain consul.securecodebox.io",
    "location": "consul.securecodebox.io",
    "name": "consul.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "target.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain target.securecodebox.io",
    "location": "target.securecodebox.io",
    "name": "target.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "vault.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain vault.securecodebox.io",
    "location": "vault.securecodebox.io",
    "name": "vault.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "docs.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain docs.securecodebox.io",
    "location": "docs.securecodebox.io",
    "name": "docs.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "nikto.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain nikto.securecodebox.io",
    "location": "nikto.securecodebox.io",
    "name": "nikto.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "api-backend.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain api-backend.securecodebox.io",
    "location": "api-backend.securecodebox.io",
    "name": "api-backend.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
  {
    "attributes": {
      "addresses": [
        {
          "asn": 24940,
          "cidr": "138.201.0.0/16",
          "desc": "HETZNER-AS ",
          "ip": "138.201.126.99",
        },
      ],
      "domain": "securecodebox.io",
      "name": "api.securecodebox.io",
      "source": "Entrust",
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain api.securecodebox.io",
    "location": "api.securecodebox.io",
    "name": "api.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`;

exports[`handles jsonl files with a single row correctly 1`] = `
[
  {
    "attributes": {
      "addresses": [
        {
          "asn": 54113,
          "cidr": "185.199.108.0/22",
          "desc": "FASTLY - Fastly",
          "ip": "185.199.109.153",
        },
      ],
      "domain": "securecodebox.io",
      "name": "www.securecodebox.io",
      "source": undefined,
      "tag": "cert",
    },
    "category": "Subdomain",
    "description": "Found subdomain www.securecodebox.io",
    "location": "www.securecodebox.io",
    "name": "www.securecodebox.io",
    "osi_layer": "NETWORK",
    "severity": "INFORMATIONAL",
  },
]
`;
